File: frontend/health_app/lib/features/visits/domain/entities/medical_visit.dart
enum MedicalVisitType { oneTime, recurring }

enum MedicalVisitSyncState { localOnly, pendingUpload, synced }

class MedicalVisit {
  const MedicalVisit({
    required this.id,
    required this.doctorName,
    required this.specialty,
    required this.appointmentDate,
    required this.timeInMinutes,
    required this.location,
    required this.visitType,
    required this.rating,
    required this.createdAt,
    required this.updatedAt,
    this.remoteId,
    this.syncState = MedicalVisitSyncState.localOnly,
  });

  final String id;
  final String doctorName;
  final String specialty;
  final DateTime appointmentDate;
  final int timeInMinutes;
  final String location;
  final MedicalVisitType visitType;
  final double rating;
  final DateTime createdAt;
  final DateTime updatedAt;
  final String? remoteId;
  final MedicalVisitSyncState syncState;

  DateTime get scheduledAt {
    return DateTime(
      appointmentDate.year,
      appointmentDate.month,
      appointmentDate.day,
      timeInMinutes ~/ 60,
      timeInMinutes % 60,
    );
  }

  MedicalVisit copyWith({
    String? id,
    String? doctorName,
    String? specialty,
    DateTime? appointmentDate,
    int? timeInMinutes,
    String? location,
    MedicalVisitType? visitType,
    double? rating,
    DateTime? createdAt,
    DateTime? updatedAt,
    String? remoteId,
    MedicalVisitSyncState? syncState,
  }) {
    return MedicalVisit(
      id: id ?? this.id,
      doctorName: doctorName ?? this.doctorName,
      specialty: specialty ?? this.specialty,
      appointmentDate: appointmentDate ?? this.appointmentDate,
      timeInMinutes: timeInMinutes ?? this.timeInMinutes,
      location: location ?? this.location,
      visitType: visitType ?? this.visitType,
      rating: rating ?? this.rating,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
      remoteId: remoteId ?? this.remoteId,
      syncState: syncState ?? this.syncState,
    );
  }

  static DateTime normalizeDate(DateTime value) {
    return DateTime(value.year, value.month, value.day);
  }
}

File: backend/HealthApp/Domain/Entity/MedicalVisit.cs
using Data.Interfaces;
using Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Domain.Entity
{
    [Table("medical_visits")]
    public class MedicalVisit : IHasId
    {
        [Key]
        public Guid Id { get; set; }
        public Guid UserId { get; set; }

        [Required]
        [MaxLength(120)]
        public string DoctorName { get; set; } = string.Empty;

        [Required]
        [MaxLength(120)]
        public string Specialty { get; set; } = string.Empty;

        public DateTime AppointmentDate { get; set; }

        [Range(0, 1439)]
        public int TimeInMinutes { get; set; }

        [Required]
        [MaxLength(255)]
        public string Location { get; set; } = string.Empty;
        public MedicalVisitType VisitType { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime LastUpdatedAt { get; set; }

        [ForeignKey(nameof(UserId))]
        [InverseProperty(nameof(Entity.User.MedicalVisits))]
        public User? User { get; set; }

        [NotMapped]
        public DateTime ScheduledAt
        {
            get
            {
                return new DateTime(
                    AppointmentDate.Year,
                    AppointmentDate.Month,
                    AppointmentDate.Day,
                    TimeInMinutes / 60,
                    TimeInMinutes % 60,
                    0);
            }
        }
    }
}
